### PR TITLE
Add disk free space cluster settings

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
Add disk free space cluster settings

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/cluster-settings.md [hunk 1]
```diff
@@ -256,6 +256,22 @@ plugins.ml_commons.jvm_heap_memory_threshold: 85
 - Default value: 85
 - Value range: [0, 100]
 
+## Set disk free space threshold
+
+Sets a disk circuit breaker that checks disk usage before running an ML task. If the disk free space exceeds the threshold, OpenSearch triggers a circuit breaker and throws an exception to maintain optimal performance.
+
+Values are in byte unit and default value is 5G, to disable this circuit breaker, set the value to 0.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:263]:
```suggestion
Valid values are in byte units. To disable the circuit breaker, set the value to 0.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @zane-neo! Is this for 2.16?

**Comment by zane-neo** [comment]
> Thank you, @zane-neo! Is this for 2.16?

Yes, for 2.16.

**Comment by zane-neo** [comment]
@kolchfa-aws can you help merge this PR?

File: _ml-commons-plugin/cluster-settings.md [hunk 2]
```diff
@@ -256,6 +256,23 @@ plugins.ml_commons.jvm_heap_memory_threshold: 85
 - Default value: 85
 - Value range: [0, 100]
 
+## Set disk free space threshold
+
+Sets a disk circuit breaker that checks disk usage before running an ML task. If the disk free space exceeds the threshold, OpenSearch triggers a circuit breaker and throws an exception to maintain optimal performance.
+
+Values are in byte unit and default value is 5G, to disable this circuit breaker, set the value to -1.
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:263]:
```suggestion
Valid values are in byte units. To disable the circuit breaker, set this value to -1.
```

**Pull request merged**